### DISCUSSION TITLE
404 intermittent spotify response when using AddItemsToPlaylistRequest

### DISCUSSION THREAD

**Discussion by cezgit** [spotify-web-api-java/spotify-web-api-java#298] [General]
Spotify's API throws an intermittent 404 on "Add Items to Playlist" requests and I've noticed that their documentation primarily suggests adding the items ids to the body, however, when using the "se.michaelthelin.spotify.requests.data.playlists.AddItemsToPlaylistRequest" (7.0.0) I see that item ids are added to the query string instead. Could this be the problem? Using the query string is probably less optimal... so I wonder if maybe using the body for the item ids would make the 404 go away...

**Comment by dargmuesli** [comment]
Instead of giving the URIs [as `string array`](https://spotify-web-api-java.github.io/spotify-web-api-java/se/michaelthelin/spotify/SpotifyApi.html#addItemsToPlaylist(java.lang.String,java.lang.String%5B%5D)), give then [as `JSONArray`](https://spotify-web-api-java.github.io/spotify-web-api-java/se/michaelthelin/spotify/SpotifyApi.html#addItemsToPlaylist(java.lang.String,com.google.gson.JsonArray)) to have them set to the body.

**Reply by cezgit** [reply]
Thank you! I will try that.
